domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentCalculation.kt

```diff
@@ -0,0 +1,105 @@
+package org.oppia.android.domain.question
+
+import org.oppia.android.app.model.FractionGrade
+import org.oppia.android.app.model.UserAssessmentPerformance
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Private mutable class that stores a grade as a fraction.
+ */
+private class MutableFractionGrade(var numerator: Int, var denominator: Int)
+
+/**
+ * Private class that computes the state of the user's performance at the end of a practice session.
+ * This class can exist across multiple training session instances, but calling code is responsible
```

Comment:
This might change if you decide to use a factory per my comment below.

However, it might also be worth clarifying here that it's the responsibility of the caller to synchronize access to this class (that is, this class makes no effort to ensure thread-safe access). I think that's already consistent with how you're using it, and it provides a nice simplification to the implementation (since there's a lot of mutable state in the current class that would need to be synchronized).

Reply:
Added

Reply:
Removed, thanks!